python/triton/compiler/code_generator.py

```diff
@@ -377,14 +373,16 @@ def visit_Return(self, node):
             ret = language.semantic.to_tensor(ret_value, self.builder)
             self.builder.ret([ret.handle])
             ret_ty = ret.type
-        # self.builder.create_branch(post_ret_block)
-        # self.builder.set_insertion_point_to_end(post_ret_block)
 
         if self.ret_type is None:
             self.ret_type = ret_ty
         elif self.ret_type != ret_ty:
             raise TypeError(f'Inconsistent return types: {self.ret_type} and {ret_ty}')
 
+        # Create a dead basic block to insert any ops that come after the return
+        post_ret_block = self.builder.create_block()
+        self.builder.set_insertion_point_to_end(post_ret_block)
```

Comment:
What ops can come after the return?

Reply:
Oh, I think it's because we don't want to create multiple returns in a single bb.

Maybe it's worth adding a comment. I actually forgot after a long time

Reply:
`tt.return` is a terminator which means it must be the last operation in a basic block, so even if we had:
```python
def fn(x):
    return x
    y = x + 1
```

Previously this would generate an invalid IR that needed to be cleaned up by the `fn.finalize()` call.